src/OrchardCore.Modules/OrchardCore.Users/Drivers/UserDisplayDriver.cs

```diff
@@ -79,11 +93,26 @@ public override async Task<IDisplayResult> UpdateAsync(User user, UpdateEditorCo
             model.UserName = model.UserName?.Trim();
             model.Email = model.Email?.Trim();
 
+            var usersOfAdminRole = await _userManager.GetUsersInRoleAsync(_administratorRole);
             if (!model.IsEnabled && user.IsEnabled)
             {
-                user.IsEnabled = model.IsEnabled;
-                var userContext = new UserContext(user);
-                await Handlers.InvokeAsync((handler, context) => handler.DisabledAsync(userContext), userContext, _logger);
+                if (usersOfAdminRole.Count == 1 && usersOfAdminRole.First().UserName == user.UserName)
+                {
+                    _notifier.Warning(H["Cannot disable the super administrator."]);
+                }
+                else
+                {
+                    if (user.UserName != httpContext.User.Identity.Name)
+                    {
+                        user.IsEnabled = model.IsEnabled;
+                        var userContext = new UserContext(user);
+                        await Handlers.InvokeAsync((handler, context) => handler.DisabledAsync(userContext), userContext, _logger);
+                    }
+                    else
+                    {
+                        _notifier.Warning(H["Cannot disable current user."]);
```

Comment:
Cannot disable current **the** user.

Reply:
I don't think it's appropriate in that order. "Cannot disable the current user" would work but you can omit using "the" in this sentence.